assets/js/theme/common/product-details.js

```diff
@@ -764,4 +765,51 @@ export default class ProductDetails {
             }
         }
     }
+
+    /**
+     * https://stackoverflow.com/a/37949642
+     */
+    replaceBulk(str, findArray, replaceArray) {
+        let stringy = str;
+        let regex = [];
+        const map = {};
+        for (let i = 0; i < findArray.length; i++) {
+            regex.push(findArray[i].replace('[-[\]{}()*+?.\\^$|#,]', '\\$0'));
+            map[findArray[i]] = replaceArray[i];
+        }
+        regex = regex.join('|');
+        stringy = stringy.replace(new RegExp(regex, 'g'), (matched) => map[matched]);
+        return stringy;
+    }
+
+    /**
+     * Buy now functionality click events
+     */
+    buyNowButton() {
+        $('body').on('click', '#form-action-validateForm', event => {
+            const $form = $(event.currentTarget).closest('[data-cart-item-add]');
+            const attr = $form.find('#form-action-addToCart').attr('disabled');
+            // For some browsers, `attr` is undefined; for others,
+            // `attr` is false.  Check for both.
+            if (typeof attr !== typeof undefined && attr !== false) {
+                const message = $form.find('.alertBox-column.alertBox-message').text();
+                showAlertModal(message);
+                return false;
+            }
+            if ($form[0].checkValidity()) {
+                const formStr = $form.serialize();
+                const replacedString = this.replaceBulk(formStr, ['%5B', '%5D', 'add', 'qty%5B%5D'], ['[', ']', 'buy', 'qty']);
+                window.location.href = `/cart.php?${replacedString}&source=buy-now-button`;
```

Comment:
as we discussed in the channel, IMO, we should be using the storefront APIs instead if this is something we want to use in cornerstone. I am fine keeping around the existent endpoint for email marketing, but for any client JS-enabled like this one, we should be using our public APIs instead.

Reply:
To be more clear, if this was not clear the first time and due to the lack of added context of my original comment. It would be ideal if we can work on top of the publicly documented APIs we have built for JS clients https://developer.bigcommerce.com/api-docs/cart-and-checkout/working-sf-apis

The `/cart.php` endpoint will likely stay around for a while, but is more limited on some capabilities and is not ideal for many scenarios. The storefront APIs are the place where we want to keep adding features and functionality. If there are some features lacking in those APIs, we should probably enhance those endpoints to reach feature parity.

The use case where I think keeping around a `GET` endpoint is mostly for email marketing purposes, where is much less common to have JS available to our advantage. For other cases, like in is in this case, where we do have access to JS capabilities, then working on top of the APIs would be ideal.

Reply:
@serroba Will work on refactoring to make use of storefront APIs. Created this a long time ago and am amazed we've rode it this far and wanted to share it with the community. Very much appreciate the feedback!

Reply:
Thank you @sacr3dc0w for your contribution and sorry for the my first comment, I didn't realize this was coming from the community, hence my lack of context in the first comment.